Path resolution docs (current state only) v2 (#11442)
ethereum/solidity: docs/using-the-compiler.rst

the remapping targets and outside of the directories where explicitly specified source
files reside, so things like ``import "/etc/passwd";`` only work if you add ``/=/`` as a remapping.

> When accessing the filesystem to search for imports, :ref:`relative paths that do not start with ./

Reviewer: ```suggestion
When accessing the filesystem to search for imports, :ref:`paths that do not start with ./
```

or ../ <relative-imports>` are treated as relative to the directory specified using
``--base-path`` option (or the current working directory if base path is not specified).
Furthermore, the part added via ``--base-path`` will not appear in the contract metadata.

For security reasons the compiler has restrictions on what directories it can access.

> Paths of source files (and their subdirectories) specified on the command line and paths defined by

Reviewer: ```suggestion
Directories of source files (and their subdirectories) specified on the command line and paths defined by
```
(Is it directories or paths?)

Author: It's directories. Changed to:

```rst
Parent directories of source files specified on the command line and target paths of remappings ...
```

I removed the part about subdirectories because I think it should be pretty intuitive. I'd expect this detail to be mentioned only if it was the opposite.

Reviewer: But if you write 'Parent directories', I would interpret this as all (super-)directories leading up to the file, i.e. if you call `solc /a/b/c/d.sol`, then everything inside `/a` is accessible.

Author: Right. Why do all the terms have to be so ambiguous!

Anyway, changed `parent directories` to `directories` now.

> remappings are automatically allowed in import statements, but everything else is rejected by default.

Reviewer: ```suggestion
remappings are automatically allowed to be accessed by the file reader, but everything else is rejected by default.
```
